_source/_posts/2020-04-20-android-authentication.adoc

```diff
@@ -0,0 +1,508 @@
+
+[#okta-oidc-config]
+== Get your Okta OIDC configuration settings
+
+Before adding anything else to your app, you'll need to quickly obtain the configuration parameters required to configure Okta OIDC authentication:
+
+1. Log in to your Okta Developer account or https://developer.okta.com/[sign up for a **free** developer account]
+2. Once you've logged on, you'll see the Admin Dashboard. Take a note of your "Org URL" (at the top right of the screen). In my case, I see: `\https://dev-123456.okta.com`
+3. Navigate to **Applications** → **Add Application**
+4. Choose **Native** as the platform
+5. Add a **Login redirect URIs** similar to `com.okta.dev-123456:/callback` (using your own `dev-XXXX`)
+6. Add a **Logout redirect URIs** similar to `com.okta.dev-123456:/logout` (using your own `dev-XXXX`)
+7. Click on **Done**
+
+image::{% asset_path 'blog/android-authentication/okta-create-new-application.jpg' %}[alt=Create a new Okta application,width=700,align=center]
+
+Next, take note of the **Client ID** that's found in the **General** tab of your new Okta application entry. It's a long string of around 20 characters.
+
+== Add a sign-in button to your Android app
+
+Now that you've finished up in Okta's Admin Panel, head back to Android Studio and add a button to your `res/layout/activity_main.xml`:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="utf-8"?>
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:fitsSystemWindows="true"
+    tools:context=".MainActivity">
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:gravity="center"
+        android:orientation="vertical">
+
+    <Button
+        android:id="@+id/signIn"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Sign in" />
+
+    </LinearLayout>
+
+</ScrollView>
+----
+
+This button will be used to authenticate using OAuth 2.0 + OpenID Connect, thanks to the Okta OIDC SDK.
+
+== Give your Android app permission to use the Internet
+
+In your `app/src/main/AndroidManifest.xml` add between the `<manifest></manifest>` tags:
+
+[source,xml]
+----
+<uses-permission android:name="android.permission.INTERNET" />
+----
+
+== Add Okta OIDC SDK to your Android app
+
+1. Add a `manifestPlaceholder` for `appAuthRedirectScheme` in `app/build.gradle`. Make sure it is consistent with your Redirect URIs. For instance, my Redirect URIs look like `com.okta.dev-123456:/callback`, therefore my `appAuthRedirectScheme` is `com.okta.dev-123456`
+
+2. Add the required Okta dependencies in `app/build.gradle`, and declare a Java 1.8 target in the `compileOptions` clause (this is required by the OIDC library)
+
+====
+[source,kotlin]
```

Comment:
This looks like Groovy, not Kotlin. If it was Kotlin, wouldn't it have a `.kt` suffix?